### PR TITLE
Update Material UI to version 5 in our React samples

### PR DESCRIPTION
**Pull request by salman90**
Material UI is not working with our local build and React 18. We will need to upgrade the following React samples:
1. [gatsby-sample](https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/samples/msal-react-samples/gatsby-sample)
2. [react-router-sample](https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/samples/msal-react-samples/react-router-sample)
3. [typescript-sample](https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/samples/msal-react-samples/typescript-sample)

To use React 18 and Material UI version 5.

### CODE REVIEW

File: samples/msal-react-samples/react-router-sample/package.json [hunk 1]
```diff
@@ -5,8 +5,10 @@
   "dependencies": {
     "@azure/msal-browser": "file:../../../lib/msal-browser",
     "@azure/msal-react": "file:../../../lib/msal-react",
-    "@material-ui/core": "^4.11.2",
-    "@material-ui/icons": "^4.11.2",
+    "@emotion/react": "^11.10.5",
```
Comment (derisen) [inline samples/msal-react-samples/react-router-sample/package.json:8]: >    "@azure/msal-browser": "file:../../../lib/msal-browser",
>    "@azure/msal-react": "file:../../../lib/msal-react"

Can we use published versions as default? (also for `react `and `react-dom` below)

File: samples/msal-react-samples/gatsby-sample/src/components/NavBar.jsx [hunk 2]
```diff
@@ -1,20 +1,18 @@
...
 import WelcomeName from "./WelcomeName";
 import SignInSignOutButton from "./SignInSignOutButton";
-import useStyles from "../styles/useStyles";
 
 const NavBar = () => {
-    const classes = useStyles();
 
     return (
-        <div className={classes.root}>
+        <div sx={{ flexGrow: 1 }}>
```
Comment (derisen) [inline samples/msal-react-samples/gatsby-sample/src/components/NavBar.jsx:12]: is `makeStyles` deprecated? why are we not getting classes from `useStyles` hook?

File: samples/msal-react-samples/react-router-sample/src/ui-components/NavBar.jsx [hunk 3]
```diff
@@ -1,20 +1,17 @@
...
 import SignInSignOutButton from "./SignInSignOutButton";
-import useStyles from "../styles/useStyles";
 import { Link as RouterLink } from "react-router-dom";
 
 const NavBar = () => {
-    const classes = useStyles();
-
     return (
-        <div className={classes.root}>
+        <div sx={{ flexGrow: 1 }}>
```
Comment (derisen) [inline samples/msal-react-samples/react-router-sample/src/ui-components/NavBar.jsx:11]: some comment about `makeStyles` here

**Review by derisen — APPROVED** [review]
minor comments, but LGTM otherwise -thanks for doing this!

**Review by tnorling — APPROVED** [review]

**Pull request merged**